G/GMP/common.jl

```diff
@@ -55,16 +56,17 @@ install_license COPYING*
 """
 
     # We enable experimental platforms as this is a core Julia dependency
-    platforms = expand_cxxstring_abis(supported_platforms(;experimental=true))
+    platforms = supported_platforms(;experimental=true)
```

Comment:
```suggestion
    platforms = supported_platforms()
```

Reply:
I thought we had agreed to cxx abi but not libgfortran?

Reply:
https://julialang.slack.com/archives/C674ELDNX/p1663017081953019?thread_ts=1663016368.261019&cid=C674ELDNX at least that's what I got from this. Because we do use libstdcxx.

Reply:
Actually, something's not working:
```
┌ Warning: /agent/_work/1/s/G/GMP/GMP@6.2.1/build/x86_64-linux-gnu-cxx03-sanitize+memory/gNsqIvnu/x86_64-linux-gnu-libgfortran3-cxx03-sanitize+memory/destdir/lib/libgmpxx.so.4.6.1 contains cxx11 ABI std::string values within its public interface, but we are supposedly building for cxx03 ABI. This usually indicates that the build system is somehow ignoring our choice of compiler, as we manually insert the correct compiler flags for this ABI choice!
└ @ BinaryBuilder.Auditor /depot/packages/BinaryBuilder/kzxHJ/src/auditor/compiler_abi.jl:258
##[warning]/agent/_work/1/s/G/GMP/GMP@6.2.1/build/x86_64-linux-gnu-cxx03-sanitize+memory/gNsqIvnu/x86_64-linux-gnu-libgfortran3-cxx03-sanitize+memory/destdir/lib/libgmpxx.so.4.6.1 contains cxx11 ABI std::string values within its public interface, but we are supposedly building for cxx03 ABI. This usually indicates that the build system is somehow ignoring our choice of compiler, as we manually insert the correct compiler flags for this ABI choice!
```

Reply:
Could clang be ignoring them?

Reply:
That's most likely, `_GLIBCXX_USE_CXX11_ABI` doesn't seem to be used much in `llvm-project`: https://github.com/llvm/llvm-project/search?q=_GLIBCXX_USE_CXX11_ABI&type=code.  So you should use the platform `Platform("x86_64", "linux"; sanitize="memory", cxxstring_abi="cxx11")`  (you can still do it before expansion, it'd be no-op).

Reply:
I wonder what's the best way to enforce this. I guess add some code to `expand_cxx_lala` that checks if it has sanitize then only add the `cxx11` abi?